server/pulp/server/content/web/views.py

```diff
@@ -68,7 +69,13 @@ def x_send(path):
         :rtype: django.http.HttpResponse
         """
         if os.access(path, os.R_OK):
-            reply = HttpResponse()
+            content_type = mimetypes.guess_type(path)
```

Comment:
FWIW, there is a Python binding to libmagic that seems to "come with" Fedora last I checked. I'm not sure about ELs though. Also, it's not documented, and I spent quite a lot of time trying to figure out how to use its API.

Oh, and there are no less than three Python libraries that are bindings to libmagic that share the same name but do not have the same API. If we depend on one of them, it will be difficult to express which one we need in our setup.py which could be bad if we ever want to work in Debian, which by the way uses a different one than Fedora uses.